src/nspawn/nspawn.c

```diff
@@ -4212,6 +4212,7 @@ static int run_container(
         ssize_t l;
         sigset_t mask_chld;
         _cleanup_close_ int netns_fd = -1;
+        _cleanup_close_ int child_netns_fd = -1;
```

Comment:
hmm, these two variables both refer to the same object btw: the child's netns. One is used in the existing code when you specify --natwork-namespace-path= and the other in any case. I think we should just reuse the same variable here, it's going to be the same thing. Moreover, if netns_fd is already found initialized (because --network-namespace-path= is set) then we shouldn't open things a second time and just reuse the thing.

Reply:
Done, but I renamed `netns_fd` to `child_netns_fd` for clarity.